docs(ECS): Added ECS documentation

BrunoCarrier opened the pull request:
docs(ECS): Added ECS documentation

bruno-carrier-lookout committed:
docs(ecs): Added ECS documentation

bruno-carrier-lookout committed:
Added ECS reference documents for the cloud provider

dorbin commented on setup/install/providers/ecs.md:
```diff
@@ -0,0 +1,115 @@
+
+### ECS Cluster
+You need to [create an ECS cluster](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/create_cluster.html) and have enough capacity in it to deploy your containers.
+Please note that ECS clusters are not the same concept as [Spinnaker clusters](/concepts/clusters/).  
```
@BrunoCarrier is it worth elaborating on the difference here?

BrunoCarrier replied:
Since the explanation is part of the glossary, I will remove this part as per your suggestion

dorbin commented on setup/install/providers/ecs.md:
```diff
@@ -0,0 +1,115 @@
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
```
Can you say more about this. Like, which yaml is this and where does it live?

BrunoCarrier replied:
Adding some context: `The role that Clouddriver assumes for your ECS account needs to have the following trust relationships for your Spinnaker IAM assumed role`

dorbin replied:
@BrunoCarrier I'd still like to see the name of the yml file and where it lives. I think it's on the ECS side, right? Not Spinnaker. But it's not clear to me.

BrunoCarrier replied:
Ah, turns out I hadn't pushed out some changes.  My apologies.  I believe this will now be clear

dorbin commented on setup/install/providers/ecs.md:
```diff
@@ -0,0 +1,115 @@
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
```
Looks like these are all edited on the Amazon Console? You might add the steps for getting to them, as is done in the [existing AWS provider setup docs](https://www.spinnaker.io/setup/install/providers/aws/).

dorbin reviewed (commented):
Thanks for the doc, Bruno!

bruno-carrier-lookout committed:
Added context to the ECS documentation

BrunoCarrier commented:
@dorbin I believe I have addressed all concerns raised.  Let me know if there is anything else I can do to make this PR mergeable! 💯 

And thanks for the rapid feedback!

bruno-carrier-lookout committed:
Amendments to PR

bruno-carrier-lookout committed:
Amendments to PR

dorbin commented on setup/install/providers/ecs.md:
```diff
@@ -0,0 +1,95 @@
+
+### Optional setups
+
+You may create IAM roles that have the `ecs-tasks.amazonaws.com` trust relationship so that your containers have an IAM role associated to them.  or information on how to modify IAM roles in the AWS console, see the [AWS documentation](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_manage_modify.html)  
```
Missing "F" at beginning of 2nd sentence :)

dorbin reviewed (approved):
Looks good, Bruno. Thanks!

bruno-carrier-lookout committed:
Amendments to PR

BrunoCarrier commented:
@dorbin I added the missing `F`

lwander commented on setup/install/providers/ecs.md:
```diff
@@ -0,0 +1,95 @@
+
+### Halyard
+
+Halyard support is in progress.  
```
Can this be updated with the recent halyard commands added by the community?

BrunoCarrier replied:
@lwander absolutely, but if you don't mind let's first merge this PR which is accurate and adds value, and then make another PR for Halyard.  Does that sound good to you?
